bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlSequence.java

```diff
@@ -637,4 +641,27 @@ private Type getSequenceType(Type tempExprType) {
                 return PredefinedTypes.TYPE_TEXT;
         }
     }
+
+    private void initializeIteratorNextReturnType() {
+        Type childrenType;
+        if (children.size() == 1) {
+            childrenType = children.get(0).getType();
+        } else {
+            Set<Type> types = new HashSet<>();
```

Comment:
Can't we use direct List here?

Reply:
The types should be added only once. So a set has been used.